synapse/events/utils.py

```diff
@@ -362,11 +362,12 @@ def serialize_event(
             e.unsigned["redacted_because"], time_now_ms, config=config
         )
 
-    if config.token_id is not None:
-        if config.token_id == getattr(e.internal_metadata, "token_id", None):
-            txn_id = getattr(e.internal_metadata, "txn_id", None)
-            if txn_id is not None:
-                d["unsigned"]["transaction_id"] = txn_id
+    if config.device_id is not None and config.device_id == getattr(
+        e.internal_metadata, "device_id", None
+    ):
```

Comment:
comments here would be helpful. Why do we need to `getattr` rather than just read `e.internal_metadata.device_id` directly?

(just because the code was under-commented before doesn't mean you need to maintain that! What is this block actually doing?)